### PR TITLE
Patch to restore position upon returning to video until more fully fleshed out fix can be introduced

### PR DESCRIPTION
**Pull request by david0178418**
"Fixes" https://github.com/OdyseeTeam/odysee-frontend/issues/20

@infinite-persistence, I'm not certain if this interfere's with what you might have in flight on this ticket, but this will patch two of the three issues until the api issue can be resolved.  Essentially, it just assumes that if there's a time stored, but there is no length known, then it assumes the stored time is correct.  Also, on dispose, if it's almost done, we'll just clear the saved time.

Of the three bugs noted, this won't address the cross-tab issue.

### CODE REVIEW

**Comment by david0178418** [comment]
Actually, after thinking about it for a second, the cross-tab bug could be patched pretty easily by just using the localStorage `storage` event to dispatch the same actions to the the other open ta bs.

File: ui/redux/actions/content.js [hunk 1]
```diff
@@ -220,6 +220,15 @@ export function savePosition(uri: string, position: number) {
     const { claim_id: claimId, txid, nout } = claim;
     const outpoint = `${txid}:${nout}`;
 
+    localStorage.setItem(
```
Comment (infinite-persistence) [inline ui/redux/actions/content.js:223]: Not sure if still relevant today, but the other localStorage usages will first check if it's available to use.  In the past, some obscure browsers didn't support it.

File: ui/store.js [hunk 2]
```diff
@@ -230,4 +230,13 @@ const store = createStore(
 const persistor = persistStore(store);
 window.persistor = persistor;
 
+window.addEventListener('storage', (e) => {
```
Comment (infinite-persistence) [inline ui/store.js:233]: Unsure if this is the best place to put it, but probably ok for now.

**Review by infinite-persistence — COMMENTED** [review]
Some minor comments

File: ui/redux/actions/content.js (hunk 1 above)
Comment (infinite-persistence) [reply ui/redux/actions/content.js:223]: It also seems like it doesn't work in iOS Safari on BrowserStack

File: ui/constants/pages.js [hunk 3]
```diff
@@ -86,3 +86,4 @@ exports.LIVESTREAM_CURRENT = 'live';
 exports.GENERAL = 'general';
 exports.LIST = 'list';
 exports.POPOUT = 'popout';
+exports.VIDEO_ALMOST_FINISHED_THRESHOLD = 0.8;
```
[the remark this answers is not in the document]
Comment (david0178418) [reply ui/constants/pages.js:89]: Browsing through the constants files, I'm not seeing one that immediately jumps out at me as a good home for this.  Is there a spot you'd suggest?

File: ui/component/viewers/videoViewer/view.jsx [hunk 4]
```diff
@@ -265,6 +266,12 @@ function VideoViewer(props: Props) {
   function onDispose(event, player) {
     handlePosition(player);
     analytics.videoIsPlaying(false, player);
+
+    const almostFinsished = player.currentTime() / player.duration() >= VIDEO_ALMOST_FINISHED_THRESHOLD;
```
[the remark this answers is not in the document]
Comment (david0178418) [reply ui/component/viewers/videoViewer/view.jsx]: Fixed.

File: ui/redux/actions/content.js (hunk 1 above)
Comment (david0178418) [reply ui/redux/actions/content.js]: Ah, right.  While localStorage is pretty widely available, it is not guaranteed to be available, depending on browser settings, such as cookie settings.

I've wrapped it in a try catch to cover this case.

File: ui/redux/actions/content.js [hunk 5]
```diff
@@ -220,10 +220,23 @@ export function savePosition(uri: string, position: number) {
...
+          claimId,
+          outpoint,
+          position,
+        })
+      );
+
+      dispatch({
+        type: ACTIONS.SET_CONTENT_POSITION,
+        data: { claimId, outpoint, position },
+      });
```
Comment (infinite-persistence) [inline ui/redux/actions/content.js:236]: Should we put the existing dispatch out of the try-catch?  If localStorage fails, at least the "regular" one still works.

**Review by infinite-persistence — CHANGES_REQUESTED** [review]

File: ui/component/viewers/videoViewer/view.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (infinite-persistence) [reply ui/component/viewers/videoViewer/view.jsx]: Typo still exists

[the remark this answers is not in the document]
Comment (david0178418) [reply ui/component/viewers/videoViewer/view.jsx]: I'm clearly having one of those mornings.  Done.

**Review by infinite-persistence — APPROVED** [review]
@tzarebczan, ready for squash.

- Re-tested after the changes, and seems to be working.
- Safari iOS doesn't seem to sync across tabs, but it's the same as today.
- `player.duration()` could be `NaN` for video that couldn't load, but the logic seems to safely resolve to `false`, so should be ok.

**Pull request merged**